common/openapi.yml

```diff
@@ -0,0 +1,124 @@
+openapi: 3.0.3
+info:
+  title: Terra User Manager
+  description: |
+    Terra User Manager
+
+  version: 0.0.1
+paths:
+  # Unauthenticated Paths
+  /status:
+    get:
+      security: [ ]
+      summary: Returns the operational status of the service
+      operationId: serviceStatus
+      tags: [ Public ]
+      responses:
+        '200':
+          description: Service can process requests
+        '500':
+          description: |
+            Service cannot process requests. That might be because dependent services are
+            unavailable, or because there is a problem with the service itself.
+
+  /version:
+    get:
+      security: [ ]
+      summary: Returns the deployed version of the service
+      operationId: serviceVersion
+      tags: [ Public ]
+      responses:
+        '200':
+          description: Version properties
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VersionProperties'
+
+components:
+  schemas:
+    # Please keep alphabetized
+    ErrorReport:
+      type: object
+      required: [message, statusCode, causes]
+      properties:
+        message:
+          type: string
+        statusCode:
+          type: integer
+        causes:
+          type: array
+          items:
+            type: string
+
+    VersionProperties:
+      type: object
+      required: [ gitTag, gitHash, github, build ]
+      properties:
+        gitTag:
+          type: string
+          description: Git tag of currently deployed app.
+        gitHash:
+          type: string
+          description: Git hash of currently deployed app.
+        github:
+          type: string
+          description: Github link to currently deployed commit.
+        build:
+          type: string
+          description: Version of the currently deployed app declared in build.gradle. Client and server versions are linked.
+
+  responses:
+    # Keep alphabetized please
+    BadRequest:
+      description: Bad request
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorReport'
+
+    Conflict:
+      description: Request conflicts with current state
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorReport'
+
+    NotFound:
+      description: Not found (or unauthorized)
```

Comment:
Maybe some endpoints dealing with user sharing for ex. could return a 404 instead of permission denied (even if the requested data does exist) to avoid leaking information?

Reply:
In general, we do not hide lack of access behind a not found return. It is a trade-off of information leak for a better user experience.

Reply:
Fair enough!